extra_stuff/lessons_in_progress/git2/git_branching.md

```diff
@@ -0,0 +1,60 @@
+# Git Branching
+
+Using Git Branch enables you to take your development path into new directions. Branching gives you the ability to represent what is typically a linear and flat development process as a multi-dimensional mind-map.
```

Comment:
I just skimmed the suggested content and after reading Kevin's comment compared it to the description of how to fork in the readme of the curriculum itself. 

> Basically, you will fork the curriculum repository, make your changes or additions to the appropriate file, then submit a pull request with your changes.

It also links to resources that closer explain the work flow. - In contrast talking about a "multi-dimensional mind-map" does sound a bit abstract.

I'm also thinking images are worth a thousand words and as far as I can tell there's currently none in there, right? - Might be worth considering.